Document/0x04f-Testing-Network-Communication.md

```diff
@@ -200,30 +200,115 @@ When both the client and server are controlled by the same organization and used
+EX: **TLS**_**AES_128**_**GCM_SHA256**
+
+**In the following, we’ll present the different algorithms of each part of the cipher suite:**
+Protocols:
+- `SSLv1`
+- `SSLv2`[rfc6176](https://tools.ietf.org/html/rfc6176)
+- `SSLv3`[rfc6101](https://tools.ietf.org/html/rfc6101)
+- `TLSv1.0`[rfc2246](https://www.ietf.org/rfc/rfc2246)
+- `TLSv1.1`[rfc4346](https://tools.ietf.org/html/rfc4346)
+- `TLSv1.2`[rfc5246](https://tools.ietf.org/html/rfc5246)
+- `TLSv1.3`[rfc8446](https://tools.ietf.org/html/rfc8446)
+
+Key Exchange Algorithms: 
+- `DSA`[rfc6979](https://tools.ietf.org/html/rfc6979)
+- `ECDSA`[rfc6979](https://tools.ietf.org/html/rfc6979)
+- `RSA`[rfc8017](https://tools.ietf.org/html/rfc8017)
+- `DHE`[rfc2631](https://tools.ietf.org/html/rfc2631) [rfc7919](https://tools.ietf.org/html/rfc7919)
+- `ECDHE`[rfc4492](https://tools.ietf.org/html/rfc4492)
+- `PSK`[rfc4279](https://tools.ietf.org/html/rfc4279)
+- `DSS`[FIPS186-4](https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.186-4.pdf)
+- `DH_anon`[rfc2631](https://tools.ietf.org/html/rfc2631) [rfc7919](https://tools.ietf.org/html/rfc7919)
+- `DHE_RSA`[rfc2631](https://tools.ietf.org/html/rfc2631) [rfc7919](https://tools.ietf.org/html/rfc7919)
+- `DHE_DSS`[rfc2631](https://tools.ietf.org/html/rfc2631) [rfc7919](https://tools.ietf.org/html/rfc7919)
+- `ECDHE_ECDSA`[rfc8422](https://tools.ietf.org/html/rfc8422)
+- `ECDHE_PSK` [rfc8422](https://tools.ietf.org/html/rfc8422) [rfc5489](https://tools.ietf.org/html/rfc5489)
+- `ECDHE_RSA` [rfc8422](https://tools.ietf.org/html/rfc8422)
+
+Block Ciphers:
+- `DES` [rfc4772](https://tools.ietf.org/html/rfc4772)
+- `DES_CBC` [rfc1829](https://tools.ietf.org/html/rfc1829)
+- `3DES` [rfc2420](https://tools.ietf.org/html/rfc2420)
+- `3DES_EDE_CBC`[rfc2420](https://tools.ietf.org/html/rfc2420)
+- `AES_128_CBC`[rfc3268](https://tools.ietf.org/html/rfc3268)
+- `AES_128_GCM` [rfc5288](https://tools.ietf.org/html/rfc5288)
+- `AES_256_CBC`[rfc3268](https://tools.ietf.org/html/rfc3268)
+- `AES_256_GCM`[rfc5288](https://tools.ietf.org/html/rfc5288)
+- `RC4_40` [rfc7465](https://tools.ietf.org/html/rfc7465)
+- `RC4_128` [rfc7465](https://tools.ietf.org/html/rfc7465)
+- `CHACHA20_POLY1305 ` [rfc7905](https://tools.ietf.org/html/rfc7905) [rfc7539](https://tools.ietf.org/html/rfc7539)
+
+Integrity Check Algorithms:
+- `MD5` [rfc6151](https://tools.ietf.org/html/rfc6151)
+- `SHA` [rfc6234](https://tools.ietf.org/html/rfc6234)
+- `SHA256` [rfc6234](https://tools.ietf.org/html/rfc6234)
+- `SHA384` [rfc6234](https://tools.ietf.org/html/rfc6234)
+
+**Remark: The efficiency of a cipher suite depends on the efficiency of its algorithms.**
+
+
+**In the following, we’ll present the recommended cipher suites to use with TLS:**
+
+IANA recommended cipher suites [rfc8447](https://tools.ietf.org/html/rfc8447#section-8) 
+[IANA_CIPHERS](https://www.iana.org/assignments/tls-parameters/tls-parameters.xhtml#tls-parameters-4):
```

Comment:
Can you be a bit more elaborate on where you get the list from and that the list presented here is a representation of now?